src/main/java/org/openml/webapplication/fantail/dc/statistical/SimpleMetaFeatures.java

```diff
@@ -11,167 +11,142 @@
+		Double PercentageOfBinaryFeatures = null;
+		Double AutoCorrelation = null;
+
+		Double NumberOfClasses = null;
+		Double MajorityClassSize = null;
+		Double MinorityClassSize = null;
+		Double MajorityClassPercentage = null;
+		Double MinorityClassPercentage = null;
+
+		// Features
+		NumberOfNumericFeatures = 0.0;
+		NumberOfSymbolicFeatures = 0.0;
+		NumberOfBinaryFeatures = 0.0;
+		for (int i = 0; i < dataset.numAttributes(); ++i) {
+			Attribute att = dataset.attribute(i);
+			if (att.isNumeric())
+				NumberOfNumericFeatures++;
+			if (att.isNominal()) {
+				NumberOfSymbolicFeatures++;
+				if (att.numValues() == 2)
+					NumberOfBinaryFeatures++;
+			}
+		}
+		PercentageOfNumericFeatures = (NumberOfFeatures > 0 ? NumberOfNumericFeatures / NumberOfFeatures * 100 : null); 
+		PercentageOfSymbolicFeatures = (NumberOfFeatures > 0 ? NumberOfSymbolicFeatures / NumberOfFeatures * 100 : null);
+		PercentageOfBinaryFeatures = (NumberOfFeatures > 0 ? NumberOfBinaryFeatures / NumberOfFeatures * 100 : null);
+
+		// MissingValues
+		NumberOfMissingValues = 0.0;
+		NumberOfInstancesWithMissingValues = 0.0;
+		for (int i = 0; i < dataset.numInstances(); i++) {
+			Instance currentInstance = dataset.get(i);
+
+			for (int j = 0; j < dataset.numAttributes(); j++) {
+				if (currentInstance.isMissing(j))
+					NumberOfMissingValues++;
 			}
-			PercentageOfNumericFeatures = (NumberOfFeatures > 0 ? NumberOfNumericFeatures / NumberOfFeatures * 100 : null); 
-			PercentageOfSymbolicFeatures = (NumberOfFeatures > 0 ? NumberOfSymbolicFeatures / NumberOfFeatures * 100 : null);
-			PercentageOfBinaryFeatures = (NumberOfFeatures > 0 ? NumberOfBinaryFeatures / NumberOfFeatures * 100 : null);
 
-			// MissingValues
-			NumberOfMissingValues = 0.0;
-			NumberOfInstancesWithMissingValues = 0.0;
+			if (currentInstance.hasMissingValue())
+				NumberOfInstancesWithMissingValues++;
+		}
+		PercentageOfInstancesWithMissingValues = (NumberOfInstances > 0 ? NumberOfInstancesWithMissingValues / NumberOfInstances * 100 : null);
+		PercentageOfMissingValues = (NumberOfFeatures * NumberOfInstances > 0 ? NumberOfMissingValues / (NumberOfFeatures * NumberOfInstances) * 100 : null);
+
+		// AutoCorrelation
+		double TimeBasedChanges = 0.0;
+		for (int i = 1; i < dataset.numInstances(); i++) {
+			Instance currentInstance = dataset.get(i);
+			Instance previousInstance = dataset.get(i - 1);
+			if (dataset.classAttribute().isNumeric()) {
+				TimeBasedChanges += Math.abs(previousInstance.classValue() - currentInstance.classValue());
+			} else if (dataset.classAttribute().isNominal()) {
+				TimeBasedChanges += (previousInstance.classValue() == currentInstance.classValue() ? 0 : 1);
+			}
+		}
+		AutoCorrelation = (NumberOfInstances > 1 ? new Double(NumberOfInstances - 1 - TimeBasedChanges) / new Double(NumberOfInstances - 1.0) : 1.0);
+
+		// Class properties
+		if (dataset.attribute(dataset.classIndex()).type() == Attribute.NUMERIC) {
```

Comment:
The characterizer assumes that the target variable is always known. This is not always the case, some datasets are on unsupervised problems, in others the target is not clear or simply unknown. The evaluation engine assumes that in those cases the last feature is always the target, but this is not always correct and leads to messy meta-features, e.g. see dataset 4133 which has a class count of 10104. We should probably not automatically assign a target class, but simply not compute supervised meta-features if the target is now known.